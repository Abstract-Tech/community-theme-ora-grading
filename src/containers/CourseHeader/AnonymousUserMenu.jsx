import React from 'react';

import { getConfig } from '@edx/frontend-platform';
import { getLoginRedirectUrl } from '@edx/frontend-platform/auth';
import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
import { Button } from '@edx/paragon';

import message from './messages';

export const getRegisterUrl = () => {
  const { LMS_BASE_URL } = getConfig();
  const locationHref = encodeURIComponent(global.location.href);
  return `${LMS_BASE_URL}/register?next=${locationHref}`;
};

export const AnonymousUserMenu = ({ intl }) => (
  <div>
    <Button
      className="mr-3"
      variant="outline-primary"
      href={getRegisterUrl()}
    >
      {intl.formatMessage(message.registerSentenceCase)}
    </Button>
    <Button
      variant="primary"
      href={`${getLoginRedirectUrl(global.location.href)}`}
    >
      {intl.formatMessage(message.signInSentenceCase)}
    </Button>
  </div>
);

AnonymousUserMenu.propTypes = {
  intl: intlShape.isRequired,
};

export default injectIntl(AnonymousUserMenu);
