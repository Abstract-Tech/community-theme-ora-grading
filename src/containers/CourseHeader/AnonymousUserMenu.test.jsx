import React from 'react';
import { shallow } from 'enzyme';

import { AnonymousUserMenu } from './AnonymousUserMenu';

jest.mock('@edx/frontend-platform', () => ({
  getConfig: () => ({
    LMS_BASE_URL: '<LMS_BASE_URL>',
  }),
}));
jest.mock('@edx/frontend-platform/auth', () => ({
  getLoginRedirectUrl: (url) => `redirect:${url}`,
}));
jest.mock('@edx/paragon', () => ({
  Button: () => 'Button',
}));

describe('Header AnonymousUserMenu component', () => {
  const props = {
    intl: { formatMessage: (msg) => msg.defaultMessage },
  };
  test('snapshot', () => {
    expect(
      shallow(<AnonymousUserMenu {...props} />),
    ).toMatchSnapshot();
  });
});
