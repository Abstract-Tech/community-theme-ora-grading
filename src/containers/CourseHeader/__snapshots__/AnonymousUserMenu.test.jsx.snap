// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header AnonymousUserMenu component snapshot 1`] = `
<div>
  <Button
    className="mr-3"
    href="<LMS_BASE_URL>/register?next=http%3A%2F%2Flocalhost%2F"
    variant="outline-primary"
  >
    Register
  </Button>
  <Button
    href="redirect:http://localhost/"
    variant="primary"
  >
    Sign in
  </Button>
</div>
`;
